assets/app.js

```diff
@@ -1,13 +1,25 @@
 // any CSS you import will output into a single css file (app.css in this case)
 import './styles/app.css';
+import 'select2';
+
+const $ = require('jquery');
 
 // Makes the "Copy to Clipboard" button work.
-document.addEventListener("DOMContentLoaded", () => {
-    const copyButton = document.getElementById('copy-button');
-    copyButton.addEventListener('click', () => {
-        const textarea = document.getElementById('text');
-        textarea.select();
-        document.execCommand('copy');
-        copyButton.innerText = copyButton.dataset.copiedText;
+$(function () {
+    // Initiate Select2, which allows dynamic entry of languages.
+    $('#lang').select2({
+        tags: true,
+        theme: 'bootstrap',
     });
+
+    // For the result page. Makes the 'Copy' button copy the transcription to the clipboard.
+    const copyButton = $('#copy-button');
+    if (copyButton.length) {
+        copyButton.on('click', () => {
+            const textarea = $('#text');
+            textarea.select();
+            document.execCommand('copy');
+            copyButton.innerText = copyButton.dataset.copiedText;
```

Comment:
`copyButton` is now a jQuery object so should be `$copyButton` and use `$copyButton.text($copyButton.data('copied-text'));`. As it is, we get "Uncaught TypeError: e.dataset is undefined".

Reply:
Done! Thanks for noticing that!